client/performance-profiler/pages/dashboard/index.tsx

```diff
@@ -2,24 +2,32 @@ import { useTranslate } from 'i18n-calypso';
 import React from 'react';
 import './style.scss';
 import DocumentHead from 'calypso/components/data/document-head';
+import { useUrlBasicMetricsQuery } from 'calypso/data/site-profiler/use-url-basic-metrics-query';
+import { useUrlPerformanceInsightsQuery } from 'calypso/data/site-profiler/use-url-performance-insights';
 import { PerformanceProfilerDashboardContent } from 'calypso/performance-profiler/components/dashboard-content';
 import { PerformanceProfilerHeader, TabType } from 'calypso/performance-profiler/components/header';
+import { LoadingScreen } from '../loading-screen';
 
 type PerformanceProfilerDashboardProps = {
 	url: string;
 	tab: TabType;
+	hash: string;
 };
 
 export const PerformanceProfilerDashboard = ( props: PerformanceProfilerDashboardProps ) => {
 	const translate = useTranslate();
-	const { url, tab } = props;
+	const { url, tab, hash } = props;
 	const [ activeTab, setActiveTab ] = React.useState< TabType >( tab );
+	const { data: basicMetrics } = useUrlBasicMetricsQuery( url, hash, true );
+	const { final_url: finalUrl, token } = basicMetrics || {};
+	const { data: performanceInsights } = useUrlPerformanceInsightsQuery( finalUrl, token );
+	const desktopLoaded = 'completed' === performanceInsights?.status;
+	const mobileLoaded = typeof performanceInsights?.mobile === 'object';
```

Comment:
If we fall into the case where desktop is loaded we are stoping the pooling on the `useUrlPerformanceInsightsQuery`,  which means that if we don't receive the mobile results when the desktop result arrives, the user be stuck in the loading screen.

Reply:
Pooling stops only when `query.state.data?.pagespeed?.status === 'completed` in https://github.com/Automattic/wp-calypso/pull/93462/files#diff-519cd42237a70ff23433f42b0773c26efd71087abf4fd44d63c78df60fba6de2R28 which means we have results for both tests.

`const desktopLoaded = 'completed' === performanceInsights?.status;` is a workaround cause `performanceInsights?.desktop` doesn't exist in the first request. @gavande1 can you fix this in the backend?